12-watchtower-API.md

```diff
@@ -127,122 +129,167 @@ This message contains all the information regarding the appointment that the cli
-		* SHOULD reject the appointment.
-	* if `op_customer_signature` is present:
-		* SHOULD reject the appointment.
-	* if `op_customer_public_key` is present:
-		* SHOULD reject the appopintment.
+* `locator` is not a `16-byte` value.
+* `start_block` is not an integer.
+* `start_block` is behind the current chain tip.
+* `end_block` is not an integer.
+* `end_block` is behind the current chain tip.
+* `dispute_delta` is not an integer.
+* `encrypted_blob` has non-feasible size.
+* `cipher` is not among the one he implements.
+* `transaction_size` is non-feasible.
 
+The receiving node SHOULD reject the appointment if:
+
+* `start_block` is too close to the current chain tip.
+* `start_block` is too far away in the future.
+* `end_block` is too far away in the future.
+* `dispute_delta` is too small.
+* `fee_rate` is too low.
+
+* if `qos` was agreed on `wt_init`:
+	The receiving node MUST also reject the appointment if:
+	* `op_customer_signature_algorithm` is missing.
+	* `op_customer_signature_algorithm` does not match with one of the supported signing algorithms.
+	* `op_customer_signature` is missing.
+	* `op_customer_public_key` is missing.
+	* `op_customer_signature` cannot be verified using `op_customer_public_key`.
+
+* if `qos` was NOT agreed on `wt_init`:
+	* The receiving node SHOULD also reject the appointment if:
+	* if `op_customer_signature_algorithm` is present.
+	* if `op_customer_signature` is present.
+	* if `op_customer_public_key` is present.
+
+The receiving node MAY accept the appointment otherwise.
+
+#### Range of values 
+
+- `start_block`: Absolute Block number 
+- `end_block`: Absolute Block number
+- `dispute_delta`: Relative Block number
+- `transaction_size`: Measured in Bytes
+- `transaction_fee`: Measured in sats
+- `cipher`: AESGCM256, CHACHA20
+- `customer_signature_algorithm`: ECDSA, SCHNORR
 
-	
 #### Rationale
+
+We can group the data fields into logical groups. 
+
+* **Appointment information**: The appointment time is defined using the `start_block` and `end_block`. WatchTower will delete the job when the appointment has expired. We recommend only using block numbers as that is the natural clock for Bitcoin. 
+* **Explicit acknowledgement of transaction details**: The `dispute_delta`, `transaction_size`, and `transaction_fee` let the WatchTower confirm the transaction is "reasonable" and it can be accepted into the blockchain (especially if there is congestion in the future). 
+* **Encrypted transaction**: The `cipher`,`encrypted_blob` states how the WatchTower can later find the dispute transaction and decrypt the justice transaction. 
+* **Customer signature** The `customer_public_key` and `customer_signature` provides an explicit message about the job from the customer.  The `customer_public_key` can also be used for refunds if applicable.
+
 The transaction `locator` can be deterministically computed by both the client and the server. Locators of wrong size are therefore invalid.
```

Comment:
should we say how?

Reply:
This was also part of your writing, and I also think is redundant at this point.

Reply:
if it is already covered somewhere, in a plain-english way, then just remove it.

Reply:
Ok